src/udiskslinuxnvmecontroller.c

```diff
@@ -292,20 +355,31 @@ udisks_linux_nvme_controller_refresh_smart_sync (UDisksLinuxNVMeController  *ctr
     }
 
   device = udisks_linux_drive_object_get_device (object, TRUE /* get_hw */);
-  g_assert (device != NULL);
+  g_warn_if_fail (device != NULL);
```

Comment:
I agree with the `g_assert` removal, but logging a warning here isn't enough, we are accessing a member of `device` below.

Reply:
Alright, there were more occurrences, changed to returning proper error (as seen elsewhere). This is a corner case that should never happen, thus it's rather safe to put more strict measures. Also, the warning is useful to be printed out just before it crashes further down the code.